### PR TITLE
Add `pr-branch-auto-delete` feature

### PR DESCRIPTION
**Pull request by bfred-it**
~~Closes~~ #1024

- [x] Add `pr-branch-auto-delete` feature
- [ ] ~~Extend `wait-for-build` to comments~~ changed my mind 😅

### CODE REVIEW

File: source/features/pr-branch-auto-delete.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		subscription.destroy();
+
+		const deleteButton = await elementFinder('.branch-action[action$="/cleanup"] [type="submit"]', 500);
+		deleteButton.click();
+	});
+}
+
+features.add({
+	id: __featureName__,
+	description: 'Automatically deletes the branch right after mergin a PR, if possible',
```
Comment (yakov116) [inline source/features/pr-branch-auto-delete.tsx]: `mergin` => `merging`

File: source/content.ts [hunk 2]
```diff
@@ -123,6 +124,7 @@ import './features/hide-readme-header.css';
 import './features/hide-obvious-tooltips.css';
 import './features/clean-discussions.css';
 import './features/clean-rich-text-editor.css';
+
```
Comment (bfred-it) [inline source/content.ts]:
```suggestion
```

**Comment by yakov116** [comment]
From #1024
> * those using long lived branches (like develop) might be able to lock down protected branches

Until someones boss smokes them....

**Comment by kidonng** [comment]
Don't want to disturb the peace here but: GitHub Mobile app implements this feature natively. So that's "Implemented by GitHub" to some degree.

**Pull request merged**